packages/core/src/core/coreToolScheduler.ts

```diff
@@ -843,17 +897,31 @@ export class CoreToolScheduler {
             }
 
             if (toolResult.error === undefined) {
+              let content = toolResult.llmContent;
+              let outputFile: string | undefined = undefined;
+              if (
+                typeof content === 'string' &&
```

Comment:
we should add an issue that we might need to support a different solution for non text output or at least throw a reasonable error message